pubber/src/main/java/daq/pubber/Pubber.java

```diff
@@ -560,6 +557,7 @@ private void processDeviceMetadata(Metadata metadata) {
       configuration.algorithm = catchToNull(() -> metadata.cloud.auth_type.value());
     }
 
+    isGatewayDevice = false;
```

Comment:
I strongly prefer either a single-assignment or final style of variable management (unless something explicitly required like in a loop).  One easy fix here would be to declare "isGatewayDevice" as final and then have an else clause for the false assignment.